node-notes-cli/app.js

```diff
@@ -0,0 +1,60 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+let parsedData;
+let notes;
+let arg;
+let arg2;
+let arg3;
+
+try {
+  arg = process.argv[2];
+  arg2 = process.argv[3];
+  arg3 = process.argv[4];
+  const data = await readFile('data.json', 'utf8');
+  parsedData = JSON.parse(data);
+  notes = parsedData.notes;
+} catch (err) {
+  console.error('Thrown Error: ', err);
+}
+
+async function create(str) {
+  const key = parsedData.nextId;
+  notes[key] = str;
+  parsedData.nextId++;
+  const jsonString = JSON.stringify(parsedData, null, 2);
+  await writeFile('data.json', jsonString);
+}
+
+async function update() {
+  notes[arg2] = arg3;
+  const jsonString = JSON.stringify(parsedData, null, 2);
+  await writeFile('data.json', jsonString);
+}
+
+async function del(str) {
+  delete notes[arg2];
+  const jsonString = JSON.stringify(parsedData, null, 2);
+  await writeFile('data.json', jsonString);
+}
+
+function read() {
+  for (const key in notes) {
+    const theKey = Number(key);
```

Comment:
It's not necessary to convert it to a Number since you are converting it right back to a string when you log it. ;-)